TTWWW-471: Prevalence Map – Map style (general)

mattga11ant opened the pull request:
Referencing this ticket: https://simonsfoundation.atlassian.net/browse/TTWWW-471

Figma design for reference: https://www.figma.com/design/wcJ1rzoLOkRaVuKp2Zae6F/Prevalence-Map?node-id=1013-11698&node-type=frame&t=Iq4mJXmQmtiojAp1-0

The final design is 100vh tall, where the map takes over a portion of that size. This ticket did not include styling for the header, filters, timeline, or info panel. As part of this task, I rearranged some of the HTML objects, and created space for them in order to properly style the map portion. The timeline and info card are commented out for now. When the header styles from task 467 are in place, the page appears at 100vh tall.

- [x] pull branch matt/TTWWW-471 and start the server locally
- [x] run npm build dev on the autism_prevalence_map folder to compile CSS and JS changes
- [x] confirm that the colors of the main map area and map SVG match the new design
- [x] confirm that the pins on the map match the new size and color within the new design
- [x] confirm that the tooltip appears when hovering a map pin, and the style matches the new design
- [ ] confirm that the map dot remains black after clicked

yjfnyc commented:
@mattga11ant I haven't really started reviewing this PR yet, but I already noticed one issue that needs to be handled in the first place. The JS code under directory `/static/autism_prevalence_map/js` should be combined into file `main.ts` under directory `/static/autism_prevalence_map/ts`. Seems to me those JS code's dependences, like `topojson.js`, `d3-array.v1.min.js` etc, also need to go to the main.ts too. Can you please work on this before I can start the review process? Thanks!
CC @roarmeow

mattga11ant commented:
@yjfnyc thank for for the initial review. I am working through the process of including the static JS files in the main.ts file. That process became a bit more time consuming, so I am still working through it. I will provide an update soon once that it complete.

mattga11ant commented:
@yjfnyc I have worked through your request to include JS dependencies from the main.ts file.

mattga11ant commented:
@yjfnyc some screenshots of this functionality locally.

<img width="1622" alt="Screenshot 2025-01-15 at 3 08 08 PM" src="https://github.com/user-attachments/assets/3a028b7b-7ed3-4d88-b552-b8fa34a23765" />
<img width="1621" alt="Screenshot 2025-01-15 at 3 08 18 PM" src="https://github.com/user-attachments/assets/46b2def0-0eb3-4d8b-bf37-bfa875003655" />

mattga11ant commented:
@yjfnyc I pushed the updated you requested, to relocated the main JS files under the TS directory, and update those files to be .ts files.

yjfnyc commented:
So far I have found the following discrepancies between the implementation with the desgin.

1. When the dot is selected, the color of the dot should be blue. now is black.
2. The height and width of the map at different breakpoint don't match the design. Please see the following screenshots as an example. The spacings to the sideways don't match the design either. 
<img width="1545" alt="Screenshot 2025-01-27 at 2 47 45 PM" src="https://github.com/user-attachments/assets/93cbf727-656d-457e-8323-4f45a0df5912" />
<img width="2289" alt="Screenshot 2025-01-27 at 2 45 43 PM" src="https://github.com/user-attachments/assets/072be49c-b1f1-4fbc-a9e0-88419923a446" />

yjfnyc reviewed (changes requested):
So far I have some code comments as well. Can you please address them before we can go further? Thanks!

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/tailwind/tailwind.css:
```diff
@@ -2,6 +2,25 @@
+  .tooltip-inner {
+    @apply p-[0.6875rem_1rem] bg-[#F4F5F6] border border-[#BFCBD1] rounded-[4px] font-serif tracking-[0.04em] text-[#585248] max-w-[200px] text-center;
+  }
+}
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/joint.ts:
```diff
@@ -0,0 +1,298 @@
+
+        if( $(window).width() < 1024 ) {
+            alert( 'This map is currently best viewed on desktop devices. A mobile-friendly version is coming soon.' );
+        }
```
This has been updated. Something like this is within the design, but we can address that when the task comes up.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/app.ts:
```diff
@@ -0,0 +1,5 @@
+    api_call_param_string: '',
+    map: {},
+    list: {},
+};
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/joint.ts:
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/app.ts:
```diff
@@ -0,0 +1,5 @@
+export const app = {
+    api_call_param_string: '',
+    map: {},
+    list: {},
```
This has been updated.

yjfnyc commented:
Thanks for the updates, @mattga11ant. It is a great progress. In terms of features, can you please address the following discrepancies between the implementation and the design?

1. Each filter component's width is fixed in the design for all the breakpoints. Can you make them fixed, so the content won't make them scale down or up, now the country filter component is quite wide and the filters wrap to 3 lines.
2. The top and bottom line of the map cannot be seen on my end.
3. As mentioned above, the width of the map at different breakpoint is different from the design.
4. There should be spacing between the map and the filters according to the design.

mattga11ant commented:
> Thanks for the updates, @mattga11ant. It is a great progress. In terms of features, can you please address the following discrepancies between the implementation and the design?
> 
> 1. Each filter component's width is fixed in the design for all the breakpoints. Can you make them fixed, so the content won't make them scale down or up, now the country filter component is quite wide and the filters wrap to 3 lines.
> 2. The top and bottom line of the map cannot be seen on my end.
> 3. As mentioned above, the width of the map at different breakpoint is different from the design.
> 4. There should be spacing between the map and the filters according to the design.
> 5. Also when hovering over the dot on the map, the dot color is black; when selected, the dot is blue. Now the behavior is the opposite.

Thank you for the note, Jinfeng.

1. I will go ahead and add fixed widths for these
2. 3. 4. This is the last of your previous notes that I have not fixed yet. The edits to the map size. I will have updates for this on Monday.
5. This was the very last commit I pushed on this branch, this has been fixed already (https://github.com/simonsfoundation/spectrum-autism-prevalence-map/pull/45/commits/40c1e650ae83282b1032ea082196710697fd6d52)

yjfnyc commented on spectrum/autism_prevalence_map/templates/autism_prevalence_map/action_buttons.html:
Please use single quotes.

mattga11ant replied:
This has been updated.

yjfnyc commented on spectrum/autism_prevalence_map/templates/autism_prevalence_map/action_buttons.html:
```diff
@@ -1,58 +1,77 @@
+    <div>
+        <a role="button" id="download-link" class="inline-block pt-2.5 pb-2.5 pl-6.5 pr-6.5 bg-med-navy rounded-sm text-sm font-bold text-white uppercase" href="#">Download Data</a>
+    </div>
+</div>
```
end of file character

mattga11ant replied:
This has been updated.

yjfnyc reviewed (changes requested):
Hi @mattga11ant thanks for making more progress, I think we can move to QA after you address the following issues I found.
1. the map width for each breakpoint is different, not a fixed width as now in code. Please see the design.
2. the gap between the map and the filter is 16px;
3. the filter min/max has some adjustment, please see Joel's comment in the filter ticket.
I also have some code comments, please if they make sense. Thanks!

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/main.ts:
```diff
@@ -0,0 +1,37 @@
+import { ttInitJoint } from './joint.ts';
+import { ttInitMap } from './map.ts';
+import { ttInitList } from './list.ts';
+import { ttInitAbout } from './about.ts';
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/main.ts:
```diff
@@ -0,0 +1,37 @@
+import { ttInitJoint } from './joint.ts';
+import { ttInitMap } from './map.ts';
+import { ttInitList } from './list.ts';
+import { ttInitAbout } from './about.ts';
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/main.ts:
```diff
@@ -0,0 +1,37 @@
+                ttInitJoint();
+                initList();
+                break;
+            case 'about':
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/about.ts:
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/tailwind.config.js:
```diff
@@ -27,30 +43,69 @@ module.exports = {
             screens: {
             },
+            height: {
+                'main': 'calc(100vh - 4.375rem)', // 100% - 70px
```
This size is no longer needed, and I have removed it. I was trying to make everything 100% the height of the viewport, but that is not possible now given the different map sizes.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/main.ts:
```diff
@@ -0,0 +1,37 @@
+                ttInitGeo();
+                ttInitProjection();
+                ttInitJoint();
+                initList();
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -0,0 +1,940 @@
+                scale = 190;
+            } else {
+                scale = 280;
+            }
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -0,0 +1,940 @@
+                .nice(d3.timeYear);        
+
+
+
```
This has been updated.

mattga11ant commented:
> Hi @mattga11ant thanks for making more progress, I think we can move to QA after you address the following issues I found.
> 
> 1. the map width for each breakpoint is different, not a fixed width as now in code. Please see the design.
> 2. the gap between the map and the filter is 16px;
> 3. the filter min/max has some adjustment, please see Joel's comment in the filter ticket.
>    I also have some code comments, please see if they make sense. Thanks!

@yjfnyc I have addressed all of your comments. Regarding the min and max filters. I set the min at 0. Within the data that I have locally, I have several data points with a max above 150. The max I have locally is 436, so I set the max at 500. It felt like a nice round number. Happy to adjust that however the team feels is best.

mattga11ant commented:
@yjfnyc one more item I forgot to mention. The map SVG within the app is proportionally not as wide as the one in Figma. For example, in Figma, when the map is at 826px x 454px, the lines of the map background touch the 4 sides. With our current SVG, at that size there is a gap on the left and right sides.

yjfnyc reviewed (changes requested):
I agree with your suggestion about the maximum value of the prevalence, you can go ahead with it because currently I am not seeing the change, assuming you wait for my confirmation?

 Can you also please position the map to where it is located(toward the right side)) in the design for each breakpoint 

I have one more comment on code.

yjfnyc commented:
> @yjfnyc one more item I forgot to mention. The map SVG within the app is proportionally not as wide as the one in Figma. For example, in Figma, when the map is at 826px x 454px, the lines of the map background touch the 4 sides. With our current SVG, at that size there is a gap on the left and right sides.

If it is convenient for you, can we have a huddle that you can tell me more about this? Thanks, @mattga11ant

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/main.ts:
```diff
@@ -0,0 +1,33 @@
+                ttInitProjection();
+                ttInitJoint();
+                ttinitList();
+                break;
```
@yjfnyc I have this update complete. The min and max filters are also updated (I just forgot to push these last night). I also noticed that the color of the map was slightly off, so I adjusted the dark-tan color.

yjfnyc reviewed (approved):
looks good to me now.
Please put comments in the ticket about the position of the map and the min/max of prevalence to remind the QA. Thanks!
